tests/lib/test_azure.py

```diff
@@ -64,3 +65,41 @@ def test_client_copying_override_options(client: Client) -> None:
         api_version="2022-05-01",
     )
     assert copied._custom_query == {"api-version": "2022-05-01"}
+
+
+@pytest.mark.parametrize(
```

Comment:
Let's add a negative test or two as well (e.g. None headers, empty headers, both retry-after and retry-after-ms headers, retry-after-ms header that is too large, retry-after-ms header that is not a number and retry-after header in different casing)